pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,51 @@
+package ipsec
+
+import (
+	"sync"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	registerMetrics sync.Once
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedMode, recordedFlavor string = "", ""
+)
+
+// RegisterIpsecMetrics in the global registry
+func RegisterIpsecMetrics() {
+	registerMetrics.Do(func() {
+		ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name: "cluster_network_ipsec_state",
+			Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+		}, []string{"mode", "api_flavor"})
+		legacyregistry.MustRegister(ipsecStateGauge)
+	})
+}
+
+func UpdateIpsecTelemetry(ipsecConfig *operv1.IPsecConfig) {
+	var flavor, mode string
+	klog.V(5).Infof("IPsec Telemetry: %v", ipsecConfig)
+	RegisterIpsecMetrics()
+	if ipsecConfig == nil {
+		mode = string(operv1.IPsecModeDisabled)
+		flavor = "4.14"
+	} else if ipsecConfig.Mode == "" {
+		mode = string(operv1.IPsecModeFull)
```

Comment:
I'd reused existing method [getIPsecMode](https://github.com/openshift/cluster-network-operator/blob/master/pkg/network/ovn_kubernetes.go#L535) to retrieve the ipsec mode.

Reply:
yes, you could just make `getIPsecMode` method as public and use it from here. I prefer to have new method for returning flavor.